README.md

```diff
@@ -70,28 +67,19 @@ pub async fn main() -> anyhow::Result<()> {
     // Provide the observability functions to the `Linker` to be made available
     // to the instrumented guest code. These are safe to add and are a no-op
     // if guest code is uninstrumented.
-    let id = adapter.lock().await.new_collector();
-    let events = add_to_linker(id, &mut linker)?;
-    let collector = Collector::new(adapter, id, events).await?;
-    let instance = linker.instantiate(&mut store, &module)?;
-
+    let trace_ctx = adapter.start(&mut linker).await?;
+    
     // get the function and run it, the events pop into the queue
     // as the function is running
-
+    let instance = linker.instantiate(&mut store, &module)?;
     let f = instance
         .get_func(&mut store, function_name)
         .expect("function exists");
-
-    OtelStdoutAdapter::start_trace(
```

Comment:
Still reading the code so maybe i'm missing this. but how does the adapter know where the start of the trace is if we remove this part of the API?

Reply:
I think this is now handled by the fact that we create a collector per module instance.. @wwkeyboard can you clarify?